docs/table_design/list_partitioning.md

```diff
@@ -0,0 +1,110 @@
+| Multiple values for each partition column in a partition | Supported. A partition can store data with different values in each partition column. In the following example, when the loaded data with  values `Los Angeles`, `San Francisco`, and `San Diego` in the the `city` column, all the data is stored in one partition.<br>`pCalifornia`.`PARTITION BY LIST (city) (    PARTITION pCalifornia VALUES IN ("Los Angeles","San Francisco","San Diego")    [, ...] )` | Not supported. A partition stores data with the same value in  the partition column.For example, the expression `PARTITION BY (city)` is used in expression partitioning. When the loaded data contains values `Los Angeles`, `San Francisco`, and `San Diego` in the `city` column, StarRocks will automatically create three partitions `pLosAngeles`, `pSanFrancisco`, and `pSanDiego`. The three partitions respectively store data with values `Los Angeles,` `San Francisco`, and `San Diego` in the `city` column. |
+| Create partitions before data loading                    | Supported. Partitions need to be created at table creation.  | No need to do so. Partitions can be automatically created during data loading. |
+| Automatically create List partitions during data loading | Not supported. If a partition corresponding to the data does not exist during data loading, an error is returned. . | Supported.If a partition corresponding to the data does not exist during data loading, StarRocks automatically creates the partition to store the data. Each partition can only contain data with the same value for the partition column. |
+| SHOW CREATE TABLE                                        | Returned the partitions definition in the CREATE TABLE statement. | After data is loaded, the statement returns the result with the partition clause used in the CREATE TABLE statement, that is, `PARTITION BY (partition_columns)`. However, the returned result does not show any automatically created partitions. If you need to view the automatically created partitions, execute `SHOW PARTITIONS FROM <table_name>`. |
+
+## Usage
+
+### Syntax
+
+```sql
+PARTITION BY LIST (partition_columns)（
+    PARTITION <partition_name> VALUES IN (value_list)
+    [, ...]
+)
+
+partition_columns::= 
+    <column> [,<column> [, ...] ]
+
+value_list ::=
+    value_item [, value_item [, ...] ]
+
+value_item ::=
+    { <value> | ( <value> [, <value>, [, ...] ] ) }    
+```
+
+### Parameters
+
+| **Parameters**      | **Parameters** | **Description**                                              |
+| ------------------- | -------------- | ------------------------------------------------------------ |
+| `partition_columns` | YES            | the name of the partition column or columns. The partition column values can be string (BINARY not supported), date or datetime, integer, and boolean values. The partition column allows `NULL` values. |
```

Comment:
![image](https://github.com/StarRocks/starrocks/assets/57167462/5dace4fa-f947-40d5-8a98-1e3f7fc95f45)
-> The names of the partition columns.